src/view-utilities.js

```diff
@@ -114,16 +112,72 @@ var viewUtilities = function (cy, options) {
     }
 
     return eles
-            .removeClass("highlighted")
-            .removeClass("unhighlighted")
+    .removeClass("highlighted")
+    .removeClass("unhighlighted")
             .removeData("highlighted"); // TODO check if remove data is needed here
-  };
+          };
 
   // Indicates if the ele is highlighted
   instance.isHighlighted = function (ele) {
     return ele.is(".highlighted:visible") ? true : false;
   };
 
+
+  //Zoom selected Nodes
+  instance.zoom = function ( eles){
+    eles.unselect();
+    cy.animate({
+      fit: {
+        eles: eles,
+        padding: 20
+      }
+    }, {
+      duration: 1000
+    });  
+    return eles;
+  };
+
+  instance.marqueeZoom = function( canvas){
+    //Make the cy unselectable
+    cy.autounselectify(true);
+    cy.elements().unselect();
+
+    var mt = new Mousetrap();
+    var shiftKeyDown = false;
+
+    mt.bind(["shift"], function() {
+      shiftKeyDown = true;
+    }, "keydown");
+
+    mt.bind(["shift"], function(){
+      shiftKeyDown = false;
+    }, "keyup");
+
+    var p_start_x, p_start_y, p_end_x, p_end_y;
+
+    cy.one('tapstart', function( event){ 
+      if( shiftKeyDown == true){
+        p_start_x = event.position.x;
+        p_start_y = event.position.y;
+      }
+    });
+
+    cy.one('tapend', function( event){
+      p_end_x = event.position.x;
+      p_end_y = event.position.y;
+      var zoomLevel = Math.min( cy.width()/ ( Math.abs(p_end_x- p_start_x)), cy.height() / Math.abs( p_end_y - p_start_y));
+      cy.animate({
```

Comment:
In order to center the selected rectangle during zoom, use pan option in the animate function instead of the position inside the zoom option. Pan option in the animate pans the top-left corner of the boundingBox of the graph to the given position value. 
Steps to be done:

-   First find the top-left point of the boundingBox of the graph. 
-   Find the distance between the center of the selected rectangle and the top-left point in both x and y coordinates.
-   Multiply these distances with zoomLevel value (Let say the resulting values are a and b (give more convenient names)).
-   In the pan option of the animation, give the position as `x: cy.width()/2 - a, y: cy.height()/2 - b`.

This should work, you do not have to give a position inside the zoom option.